tools/rpmkeys.cc

```diff
@@ -42,19 +44,59 @@ static struct poptOption optionsTable[] = {
     POPT_TABLEEND
 };
 
-static ARGV_t gpgkeyargs(ARGV_const_t args) {
-    ARGV_t gpgargs = NULL;
-    for (char * const * arg = args; *arg; arg++) {
-	if (strncmp(*arg, "gpg-pubkey-", 11)) {
-	    char * gpgarg = NULL;
-	    rstrscat(&gpgarg, "gpg-pubkey-", *arg, NULL);
-	    argvAdd(&gpgargs, gpgarg);
-	    free(gpgarg);
-	} else {
-	    argvAdd(&gpgargs, *arg);
+static int matchingKeys(rpmKeyring keyring, ARGV_const_t args, void * userdata, int callback(rpmPubkey, void*))
+{
+    int ec = EXIT_SUCCESS;
+    if (args) {
+	for (char * const * arg = args; *arg; arg++) {
+	    int found = false;
+	    auto iter = rpmKeyringInitIterator(keyring, 0);
+	    rpmPubkey key = rpmKeyringIteratorNext(iter);
+	    while (key) {
```

Comment:
Just do

```
while ((key = rpmKeyringIteartorNext(iter))) {
    ...
}
```

That avoids having to call it twice - and is what everything else in the codebase does.

Reply:
Other than that, looks good enough for government work as they say.

Since the tests for the iterator are in this PR, we basically need to have it all in a single PR. So close one of these and update the couple of remarks in the other, and we're good to go.